python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -98,7 +98,6 @@ def dist_inference(g, gnn_encoder, node_feats, batch_size, fanout,
     dict of Tensor : the final GNN embeddings of all nodes.
     """
     device = gnn_encoder.device
-    x = node_feats
     with th.no_grad():
```

Comment:
The reason you want to compute input embedding within the for loop is that node_feats consumes many memory and we can not release it.

But if we do x = compute_node_input_embeddings(), x will automatically be released in L152.

Reply:
this approach requires to keep at least two embedding tensors of all nodes in memory.

Reply:
We always need two tensors (one for N-1 layer and one for N), right?